### PR TITLE
add extended crrcascade tests

### PR DESCRIPTION
**Pull request by SylvainSenechal**
Issue: ZENKO-5311

Second pr for crr cascade tests.
Will probably create a third, I think there are 2 new cases worth testing.

Adding 2 more important tests :
- working with non empty object, as they are handled quite differently in backbeat
- adding test focusing on metadata-tag updates, which is where we really see if microVersionId is doing its job

Tested in my codespace 🫡  
image

Also running "yarn upgrade" to close dependabot opened prs

### CODE REVIEW

File: tests/functional/ctst/features/replication/crrCascade.feature [hunk 1]
```diff
@@ -40,6 +40,46 @@ Feature: CRR Cascade Replication
         When I wait 15 seconds
         Then the cascade replication states should be settled
 
+    @2.15.0
+    @PreMerge
+    @ReplicationTest
+    @CRRCascade
+    Scenario: Cascade replication loop with non-empty object : A -> B -> C -> A
```
Comment (SylvainSenechal) [inline tests/functional/ctst/features/replication/crrCascade.feature]: actually could be merged with another test and just configure 2 example : one with empty object, another with object with some data

File: tests/functional/ctst/features/replication/crrCascade.feature [hunk 2]
```diff
@@ -40,6 +40,46 @@ Feature: CRR Cascade Replication
         When I wait 15 seconds
         Then the cascade replication states should be settled
 
+    @2.15.0
```
Comment (delthas) [inline tests/functional/ctst/features/replication/crrCascade.feature]: All those new tests should be marked 2.16 since that's where CRR will land.

**Review by maeldonn — COMMENTED** [review]
Need a rebase 😬

File: tests/functional/ctst/features/replication/crrCascade.feature [hunk 3]
```diff
@@ -25,21 +25,47 @@ Feature: CRR Cascade Replication
...
         Then the object should replicate to location "crr-location-b" within 300 seconds
         And the object should replicate to location "crr-location-c" within 300 seconds
         And the object at location "crr-location-a" should never have replication status PENDING within 30 seconds
         When I wait 15 seconds
         Then the cascade replication states should be settled
 
+        Examples:
+            | description      | objectName                | bodySize |
+            | empty object     | cascade-loop-obj          | 0        |
+            | non-empty object | cascade-loop-nonempty-obj | 42       |
```
Comment (SylvainSenechal) [inline tests/functional/ctst/features/replication/crrCascade.feature:46]: It's important to test empty vs non empty : The backbeat logic is can be quite different depending on that (no putData)

File: tests/functional/ctst/features/replication/crrCascade.feature [hunk 4]
```diff
@@ -25,21 +25,47 @@ Feature: CRR Cascade Replication
...
+
+    @2.16.0
+    @PreMerge
+    @ReplicationTest
+    @CRRCascade
+    Scenario: Cascade tag update replicates through a loop without bouncing back : A -> B -> C -> A
```
Comment (SylvainSenechal) [inline tests/functional/ctst/features/replication/crrCascade.feature:52]: This is also important : All previous tests are creating and working with fresh objects and just checking object creation/replication. These paths don't really test what happens when we don't create object, but instead perform operations on existing object metadata without creating new versions, which is where microVersionID comparisons are more involved

Here, by adding tags, we have better tests around the microVersionId update and comparison

**Review by maeldonn — APPROVED** [review]
Well done ! 👍

File: tests/functional/ctst/steps/crrCascade.ts [hunk 5]
```diff
@@ -67,27 +69,70 @@ async function putCascadeObject(
...
+
+When('tags are put on the object {string} in location {string}',
+    async function (this: Zenko, objectName: string, location: string) {
+        const cascadeBuckets = this.getSaved<Record<string, string>>('cascadeBuckets');
+        const tagValue = Utils.randomString().toLowerCase();
+        Identity.useIdentity(IdentityEnum.ACCOUNT, location);
+        await this.createS3Client().send(new PutObjectTaggingCommand({
+            Bucket: cascadeBuckets[location],
+            Key: objectName,
+            Tagging: { TagSet: [{ Key: 'cascade-test-tag', Value: tagValue }] },
```
Comment (SylvainSenechal) [inline tests/functional/ctst/steps/crrCascade.ts:103]: Reading this again, I'm thinking i could've made the tag value a parameter
oh well, its fine but i can also change it

Comment (SylvainSenechal) [reply tests/functional/ctst/steps/crrCascade.ts:103]: I think I'll do it with the next pr to not rerun everything

**Review by benzekrimaha — APPROVED** [review]
